app/templates/listen.hbs

```diff
@@ -46,35 +46,42 @@
   </div>
 
   <div class="playlist-history">
-    <div class="currently-listening-to">
-      <div class="currently-listening-to-label">CURRENTLY LISTENING TO</div>
-        <div class="song-metadata currently-playing-song">
-          <div class="dot"></div>
-            <div class="metadata">
-            <div class="composer">{{model.stream.currentPlaylistItem.catalogEntry.composer.name}}</div>
-            <div class="title_expand">
-              <NyprAToggleBox class="secondary-metadata-toggle-box" as |toggle|>
-                <toggle.trigger>
-                  <span class="title">{{model.stream.currentPlaylistItem.catalogEntry.title}}</span>
-                </toggle.trigger>
-                <toggle.dropdown>
-                  <div class="secondary-metadata-content">
-                    {{#if model.stream.currentPlaylistItem.catalogEntry.ensemble.name}}
-                      <div class="secondary-metadata-content-field"><span class="secondary-metadata-content-label">Performed by the </span><span class="secondary-metadata-content-title">{{model.stream.currentPlaylistItem.catalogEntry.ensemble.name}}</span></div>
-                    {{/if}}
-                    {{#if model.stream.currentPlaylistItem.catalogEntry.conductor.name}}
-                      <div class="secondary-metadata-content-field">
-                        <span class="secondary-metadata-content-label">Conducted by </span>
-                        <span class="secondary-metadata-content-title">{{model.stream.currentPlaylistItem.catalogEntry.conductor.name}}</span></div>
-                    {{/if}}
-                  </div>
-                </toggle.dropdown>
-              </NyprAToggleBox>
+    {{#if model.stream.currentPlaylistItem.catalogEntry}}
+      <div class="currently-listening-to">
+        <div class="currently-listening-to-label">CURRENTLY LISTENING TO</div>
```

Comment:
Since this has quite a bit of template logic in it, this whole section should be abstracted to its own component so it can be easily tested. That talk we had the other day about the service that would have properties on it for the currently playing track—this could tie in nicely with that

```javascript
{{#if model.stream.currentPlaylistItem.catalogEntry }}
  <CurrentlyPlayingTrack
      @trackTitle={{this.currentStream.trackTitle}}
      @showTitle={{this.currentStream.showTitle}}
      @episodeTitle={{this.currentStream.episodeTitle}}
      @ensembleName={{this.currentStream.ensembleName}}
      @conductorName={{this.currentStream.conductorName}}
      @composerName={{this.currentStream.composerName}}
   />
{{/if}}```

Reply:
Good catch -- I'm doing this in https://github.com/nypublicradio/nypr-jukeboxes/pull/22, which I plan on finishing up today/tomorrow.